### ISSUE TITLE
Android notification settings not respected

### ISSUE THREAD

**Issue by licaon-kter** [redsolution/xabber-android#892]
HEAD fdroid branch, Huawei Pie

Steps:
* two public MUCs (current XEPs, not future yet not even submitted ones)
* unable to setup "notify only on mention" since that's not a feature
* try custom notifications
* setup via Android Settings to "block" for each

Expected behaviour:
* never see notifications (it's either ALL or NONE for Xabber atm)

Actual result:
* random notifications popping up for both

**Comment by andrewnenakhov** [comment]
Could you please describe steps with more clarity? Also, not exactly clear what you want to achieve. Get notified only whey you are mentioned?

btw, you may call MUCs as just MUCs, we use a different name for our upcoming group chats (they are called, 'Group chats', or 'Awesome group chats' or 'new awesome group chats').

**Comment by licaon-kter** [comment]
> Get notified only whey you are mentioned?

I'd like that, but apparently you, for unknown reasons, [tie this **UI feature**](https://github.com/redsolution/xabber-android/issues/156#issuecomment-453754286) to _"the awesome not yet submitted, hopefully translated by now, groupchats of the future"_.

So, now I need to decide, do I mute Xabber completely or can I mute those groups specifically? Looking in the group settings I see something that sounds useful _"custom notifications"_... so I toggle _"Use custom..."_

Unbeknown to me (UI anti-pattern?) a _magical_ setting appears _elsewhere_... so when annoyed by these notifications that would not stop, I stumble, yet again in the apps notifications settings only to see that now, each group has its own entry, niiice, so I can just "disable" them instead of muting the whole app. Victory!

Or so I thought...

**Comment by andrewnenakhov** [comment]
@licaon-kter can't you just press 'mute' on a MUC? We didn't do an obvious feature to notify on mention for a simple reason: mentions are not defined in MUCs, unless you count 'somename,' as mention. Or was it 'somename:'? or both?

Please be more clear with your descriptions. It's really hard to follow what you mean by some things, like 'a magical setting appears elsewhere...' - which setting do you call 'magical' ? 

Anyway, our new group chats have [insanely great mentions](https://xmpp.redsolution.com/upload/4bddf4f264f5c6577f16551f16a0abdf3f7ff84d/4LuK55rm/%D0%B8%D0%B7%D0%BE%D0%B1%D1%80%D0%B0%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5.png), editable, selectable, etc. We really are not keen on dragging the dead weight of supporting obsolete standards to shiny new feauture, so current 'muc' is almost an afterthought.

**Comment by licaon-kter** [comment]
> unless you count 'somename,' as mention

That would be enough for me, yes.

> lease be more clear with your descriptions. It's really hard to follow what you mean by some things, like 'a magical setting appears elsewhere...' - which setting do you call 'magical' ?

Enabling "custom notifications" in a group chat **creates a new notification channel**.
How/When does a user get informed about that? I did not see a toast or anything that could signal that "NOW you can do this, but you need to go here".
eg. many apps have delegated notifications settings to Android since 8 brought channels, so you can put a button there that opens _"this specific channel for this group"_ so the user can tweak it.

> Anyway, our new group chats...

Are nowhere to be found, no XEP about them, no server source that one can compile and host, so no one can create/use them, yes... ok, great, we'll see them when they get here. This is not about those. This is just about _Android notifications_ and afaik, there's no XEP for those, that's just Android UI bits that need to be tamed and if-then-else them to not-show a notification IF show=disabled, or something.

>  can't you just press 'mute' on a MUC?

I can... I did it now, let's see if it does help. But, this, while _(in theory, I'll have to see if it doesn't have any other issues)_ doing what I want _(aka mute it)_ is not an actual fix for this Android issue at hand.

**Comment by andrewnenakhov** [comment]
> Enabling "custom notifications" in a group chat creates a new notification channel.

Yeah, right. Just like any Telegram does. That's the only good way now in new androids

> How/When does a user get informed about that? I did not see a toast or anything that could signal that "NOW you can do this, but you need to go here".

Why on earth should a user ever be informed about such things? If google wanted a user to be informed about the creation of new notification channels, it could have made a requirement for a user to explicitly grant permission for it. Just like for Location/Camera/etc.

> eg. many apps have delegated notifications settings to Android since 8 brought channels, so you can put a button there that opens "this specific channel for this group" so the user can tweak it.

This would lead to extremely shitty user experience. How do you even explain to a user they should go to some system notification setting and manage notifications over there? A good number of Xabber users DID 'manage' their notifications (like changing the priority of persistent notification leading to Xabber ringing non-stop on every reconnect), and It's a catastrophe. 

> Are nowhere to be found, no XEP about them, no server source that one can compile and host

They are available built-in in [Xabber Server](https://github.com/redsolution/xabber-server) source code of which is public. Installation instruction is [here](https://github.com/redsolution/xabber-server/wiki). Anyone can create and use them, either on own hosted server or on *gc.xabber.com* using the development version of [Xabber for Web](https://web.xabber.com/develop/), using any XMPP account. Just put *gc.xabber.com* into custom server field ([see pic](https://xmpp.redsolution.com/upload/4bddf4f264f5c6577f16551f16a0abdf3f7ff84d/xALA3njs/%D0%B8%D0%B7%D0%BE%D0%B1%D1%80%D0%B0%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5.png)) (btw it also has a mostly working voip!) (and screen sharing!!)

> But, this, while (in theory, I'll have to see if it doesn't have any other issues) doing what I want (aka mute it) is not an actual fix for this Android issue at hand.

To be honest, this issue is not at hand for us at all. We have much more pressing issues that have way higher priority. We know our goal. Great group chat experience is part of that goal. Good notifications are necessary for group chats. We'll do it when we'll do it, and be assured that we are moving towards that goal as fast as we can. We just have some more plans on our roadmap before that goal.

**Comment by licaon-kter** [comment]
Muting does what I want for now.

Hope these notifications get fixed too. ;)
